web_src/js/features/comp/Paste.js

```diff
@@ -124,17 +124,19 @@ async function handleClipboardImages(editor, dropzone, images, e) {
   }
 }
 
-function handleClipboardText(textarea, text, e) {
-  // when pasting links over selected text, turn it into [text](link), except when shift key is held
-  const {value, selectionStart, selectionEnd, _shiftDown} = textarea;
-  if (_shiftDown) return;
+function handleClipboardText(textarea, e, text, isShiftDown) {
```

Comment:
```suggestion
function handleClipboardText(textarea, e, text, {isShiftDown} = {}) {
```

Better to have options argument.

Reply:
I do not think it's worth to have an option here. It is not optional.

Reply:
Could remove the ` = {}` to make it required, but I guess current 4 args is borderline acceptable. Still I prefer options arg because it's immediately clear what it does. Your variant is a [boolean trap](https://ariya.io/2011/08/hall-of-api-shame-boolean-trap).

Reply:
1. I do not see any difference between `handleClipboardText(el, text, e, isShiftDown)` vs ``handleClipboardText(el, text, e, {isShiftDown})` for callers. `{}` doesn't help readability.
2. I am just removing the legacy `el._shiftDown`, and it doesn't really cause "boolean trap" because it only uses variables, no `true/false` trick.

Reply:
Actually, I am the man who keeps removing the "boolean trap" from Gitea's codebase:

![image](https://github.com/go-gitea/gitea/assets/2114189/e4d2cd11-f945-4586-a38c-f8ee126fe5be)

Reply:
And this:

![image](https://github.com/go-gitea/gitea/assets/2114189/e30d450f-d88c-41a0-a09a-7cfe8bf4166f)

Reply:
Ah yes, if you still have the variable name at caller site, that's fine.